pkg/container-collection/k8s.go

```diff
@@ -181,3 +192,42 @@ func (k *K8sClient) ListContainers() (arr []Container, err error) {
 	}
 	return arr, nil
 }
+
+func getCurrentKubeletConfig(clientset *kubernetes.Clientset, nodeName string) (*kubeletconfig.KubeletConfiguration, error) {
+	resp, err := clientset.CoreV1().RESTClient().Get().Resource("nodes").Name(nodeName).Suffix("proxy", "configz").DoRaw(context.TODO())
```

Comment:
How does this `Suffix("proxy", "configz")` work? Would the API server contact the Kubelet to fetch its config or is it cached in the node resource?

Do you know if this API is documented somewhere?

I wonder what are the pros and cons of this code compared to looking at the node resource
```
  daemonEndpoints:
    kubeletEndpoint:
      Port: 10250
```
and contact the kubelet directly. I guess your method will work even when `ig` runs in a pod in its own network namespace?

Reply:
The kubelet API is properly documented but the /configz endpoint is mentioned in the documentation in a few places, like: https://k8s-docs.netlify.app/en/docs/tasks/administer-cluster/reconfigure-kubelet/#generate-the-configuration-file

Going through the API server solves problems around auth I stumbled upon when I tried to connect directly to kubelet.

What I'm doing here is equivalent to:
```
kubectl get --raw "/api/v1/nodes/dorserero-dev/proxy/configz"
```

And if we were to connect to kubelet directly it's:
```
curl --insecure -v https://localhost:10250/configz
```
Assuming the kubelet endpoint port is 10250.

Since this happens only once, proxying the request through the API server sounds reasonable to me.

Reply:
I would like to know which version of Kubernetes introduced this `proxy/configz`.
If it is recent, we should document it on:
* docs/getting-started/requirements.md
* https://inspektor-gadget.io/docs/latest/getting-started/requirements/

Reply:
I looked specifically at 1.26 as this is the oldest supported version at this time. 
The `/api/v1/nodes/{name}/proxy/{path}` endpoint exists in [1.26](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/).

The `/configz` endpoint is undocumented but there's this comment: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/configz/OWNERS

As well as this commit where it was first introduced:
https://github.com/kubernetes/kubernetes/pull/20528